packages/hardhat-core/test/internal/solidity/resolver.ts

```diff
@@ -594,3 +594,33 @@ describe("Resolver regression tests", function () {
     });
   });
 });
+
+describe("TASK_COMPILE: the file to compile is trying to import a directory", function () {
+  useFixtureProject("compilation-import-folder");
+  useEnvironment();
+
+  it("should throw an error because a directory is trying to be imported", async function () {
+    await expectHardhatErrorAsync(
+      async () => {
+        await this.env.run(TASK_COMPILE);
+      },
+      ERRORS.RESOLVER.INVALID_IMPORT_OF_DIRECTORY,
+      "HH414: Invalid import some-plugin from contracts/A.sol. Attempting to import a directory. Directories cannot be imported."
+    );
+  });
+});
+
+describe("TASK_COMPILE: the file to compile is trying to import a non existing file", function () {
```

Comment:
I'm not against adding this if we didn't have such a test, but just to make sure I understand: this is unrelated to the PR changes, right?

Reply:
Correct